modules/ballerina-core/src/main/java/org/ballerinalang/model/types/BMapType.java

```diff
@@ -33,20 +33,29 @@
  */
 public class BMapType extends BType implements BIndexedType {
 
-    private BType elementType;
+    private BType constraint;
 
     /**
      * Create a type from the given name.
      *
      * @param typeName string name of the type
      */
-    BMapType(String typeName, BType elementType, String pkgPath) {
+    public BMapType(String typeName, BType constraint, String pkgPath) {
```

Comment:
Please update the java doc as well. Btw, elementType and pkgPath weren't in JavaDoc. Can you add it. ?

Reply:
+1